### PR TITLE
OCPBUGS-38573: use pooled client for etcd single member health checks

### PR DESCRIPTION
**Pull request by tjungblu**
This should reduce CPU consumption because of less TLS handshakes and makes it usable from other places too.

### CODE REVIEW

File: pkg/etcdcli/health.go [hunk 1]
```diff
@@ -80,30 +80,17 @@ func GetMemberHealth(ctx context.Context, etcdMembers []*etcdserverpb.Member) me
...
-	}
-	cli, err := newEtcdClientWithClientOpts([]string{member.ClientURLs[0]}, skipConnectionTest)
-	if err != nil {
-		return healthCheck{
-			Member:  member,
-			Healthy: false,
-			Error:   fmt.Errorf("create client failure: %w", err)}
-	}
+func checkSingleMemberHealth(ctx context.Context, cli *clientv3.Client, member *etcdserverpb.Member) healthCheck {
+	originalEndpoints := cli.Endpoints()
```
Comment (dusk125) [inline pkg/etcdcli/health.go:84]: nit: You could do:
```go
defer cli.SetEndpoints(cli.Endpoints())
```
and cli.Endpoints() will be cached and that cached version will be used at deferred execution

**Review by dusk125 — COMMENTED** [review]
/lgtm

File: pkg/etcdcli/health.go (hunk 1 above)
Comment (tjungblu) [reply pkg/etcdcli/health.go:84]: I leave this for now to not have your lgtm being wasted - thanks for the research, I was curious about this too :)

**Comment by tjungblu** [comment]
/cherry-pick release-4.17 release-4.16 release-4.15 release-4.14

**Comment by tjungblu** [comment]
/override ci/prow/e2e-aws-ovn-etcd-scaling

that's a known failure 
> clusteroperator/kube-storage-version-migrator should not change condition/Available

File: pkg/etcdcli/health.go (hunk 1 above)
Comment (dusk125) [reply pkg/etcdcli/health.go:84]: No worries! Here's the proof! https://go.dev/play/p/Cgzi2FAO45G

The great package github.com/pkg/profile abuses this to make profiling easier to set up.

**Pull request merged**